### DISCUSSION TITLE
DB Function does not exist? Even though I can call directly from SQL Editor?

### DISCUSSION THREAD

**Discussion by LukeEightfold** [supabase/supabase#14068] [Questions] [answered]
When I call a function from within an RLS policy, Supabase says it does not exist. 

But I can call the function directly from SQL editor and it works fine. 

What are the possible causes of this error?  See screenshots.

**1. **Calling function directly - Successful****
<img width="846" alt="Screen Shot 2023-05-02 at 4 01 10 pm" src="https://user-images.githubusercontent.com/62526367/235591871-60c8a7f5-65c7-4bd7-97bd-633045565c24.png">
**2. Error when trying to call from RLS policy - Error shown**
<img width="1191" alt="Screen Shot 2023-05-02 at 4 00 09 pm" src="https://user-images.githubusercontent.com/62526367/235591876-84fcb01b-a655-42d6-8cd2-e1bd5b8fec19.png">

**Comment by GaryAustin1** [comment]
You don't show your function definition.  Is the 2nd parameter defined as string or UUID?  In the RLS case you are passing in a UUID type so the parameter must be declared UUID or you need to cast uid()::text.  Also you are passing in a table row of bigint type evidently.  Is your first parameter declared bigint?

**Reply by LukeEightfold** [reply]
Here is function definition. 

<img width="672" alt="Screen Shot 2023-05-03 at 8 16 25 am" src="https://user-images.githubusercontent.com/62526367/235797872-60c90fae-d7fe-4e25-9474-333653f82f22.png">

The 2nd parameter is defined as uuid. 

The 1st parameter is defined as 'integer' which isn't a bigint. 

So I created another function with bigint as the input - I get the same error message. 

<img width="668" alt="Screen Shot 2023-05-03 at 8 20 48 am" src="https://user-images.githubusercontent.com/62526367/235800399-14ed7751-362d-403c-9f45-848549a0be47.png">
<img width="948" alt="Screen Shot 2023-05-03 at 8 32 09 am" src="https://user-images.githubusercontent.com/62526367/235800411-a8d2961d-7990-4563-9f4d-c11d7e506d89.png">
<img width="1178" alt="Screen Shot 2023-05-03 at 8 34 03 am" src="https://user-images.githubusercontent.com/62526367/235800542-f43a57f5-931e-44eb-ac35-6e7c5175dc61.png">
<img width="665" alt="Screen Shot 2023-05-03 at 8 34 54 am" src="https://user-images.githubusercontent.com/62526367/235800456-f6fba13d-a795-489a-af51-d9a719e6b339.png">

**Reply by GaryAustin1** [reply]
You used a capital letter in your new function which you should not do in Postgres.  https://github.com/orgs/supabase/discussions/12893
You now have to put the name in double quotes when you use it as bigInt_test gets converted to bigint_test which does not match.

**Reply by LukeEightfold** [reply]
Okay changed to lowercase. Now get different error. Is this to do with return type of function being a record?

<img width="1204" alt="Screen Shot 2023-05-03 at 8 49 14 am" src="https://user-images.githubusercontent.com/62526367/235801973-36a7eeb1-fa1d-4b12-9f4b-b250cbe2462c.png">

**Reply by GaryAustin1** [reply, marked as answer]
Probably.  You probably should just return true or false as the result (boolean).   I would also move auth.uid() inside the function if you are always calling from RLS as then it is also protected from someone using feeding in fake UUIDs.

**Reply by LukeEightfold** [reply]
For future users this has been solved. 
- The original issue was using incorrect input types (not using int8 / bigint) when passing in an account_id above. 
- Make sure you don't use camel case as mentioned above. 
- Make sure that when using IN the return type matches. In this case I was returning type record, but trying to match against a UID. Once I set return type of function to UID it worked.  

Thanks @GaryAustin1! (and Andrew Smith on support)
